src/power_grid_model/utils.py

```diff
@@ -301,3 +304,63 @@ def _compatibility_deprecated_import_json_data(json_file: Path, data_type: Datas
         raise PowerGridSerializationError("An internal error occured during deserialization")
 
     return result
+
+
+def self_test():
+    """
+    Perform a self-test of the Power Grid Model functionality to check if the installation was successful
+    and there are no build errors, segfaults, undefined symbols, etc.
+
+    This function is designed to validate the basic functionality of data serialization,
+    model instantiation, power flow calculation, and serialization of results using the
+    Power Grid Model library.
+
+    Raises:
+        PowerGridSerializationError: if an internal error occured during deserialization.
+        PowerGridError: if there was an internal error.
+    """
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create a simple JSON input data file in the temporary directory
+        input_data = {
+            "version": "1.0",
+            "type": "input",
+            "is_batch": False,
+            "attributes": {},
+            "data": {
+                "node": [{"id": 1, "u_rated": 10000}],
+                "source": [{"id": 2, "node": 1, "u_ref": 1, "sk": 1e20}],
+                "sym_load": [{"id": 3, "node": 1, "status": 1, "type": 0, "p_specified": 0, "q_specified": 0}],
+            },
+        }
+
+        input_file_path = Path(temp_dir) / "input_data.json"
+        input_file_path.write_text(json.dumps(input_data))
+
+        try:
+            # Load the created JSON input data file (deserialize)
+            deserialized_data = json_deserialize_from_file(input_file_path)
+
+            if get_dataset_type(deserialized_data) != "input":
+                raise PowerGridSerializationError("An internal error occured during deserialization")
+
```

Comment:
probably not necessary. it'll crash on the creation of the `PowerGridModel` instance.
```suggestion
```
If you do want to do this, you can also query the input itself, e.g.:
```suggestion
            assert deserialized_data[ComponentType.node]["u_rated"] == 10000
```
`AssertionError` is OK here because you will `except Exception` and rethrow it later on anyways